Task 4: 
- Identify if there are any restaurant chains present in the dataset.
- Analyze the ratings and popularity of different restaurant chains

In [2]:
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns

# Load the data
df = pd.read_csv('/Users/tilakpatel/Personel/Cognifyz intern/code/Data/Dataset.csv', encoding='utf-8')
df = df.dropna(subset=['Cuisines'])

In [14]:
df_res_chains = df.copy()
df_chains = df_res_chains.groupby('Restaurant Name')['Aggregate rating'].agg(['count', 'mean']).sort_values(by='count', ascending=False)

In [22]:
df_chains

count      mean
Restaurant Name                                         
Cafe Coffee Day                             83  2.419277
Domino's Pizza                              79  2.740506
Subway                                      63  2.907937
Green Chick Chop                            51  2.672549
McDonald's                                  48  3.339583
...                                        ...       ...
Ghungroo Club & Bar - By Gautam Gambhir      1  3.800000
Ghar Ki Handi                                1  3.000000
Ghar Ka Swad                                 1  0.000000
Ghar Bistro Cafe                             1  4.400000
�ukura��a Sofras۱                            1  4.400000

[7437 rows x 2 columns]

**Branch Count (Popularity)**:

- Cafe Coffee Day (83 branches) is the most widespread, followed by Domino's Pizza (79), and Subway (63).
- McDonald's, with 48 branches, has fewer outlets but stands out in ratings.

**Ratings (Mean)**:

- McDonald's has the highest rating among large chains (3.34), indicating consistent quality.
- Cafe Coffee Day has the most branches but a relatively low rating (2.42), suggesting room for improvement.
- Domino's Pizza and Subway have average ratings (2.74 and 2.91, respectively).

**Small Chains**:
- Smaller chains like Ghar Bistro Cafe and 'ukura Sofras۱ have high ratings (4.4), likely due to localized excellence.

**Insights**:
- Large chains often trade quality for scalability, while smaller chains maintain higher satisfaction levels. Notable exceptions include McDonald's, which balances both well.